Create a new pull request by comparing changes across two branches

GulajavaMinistudio opened the pull request:
<!-- What's in this pull request? -->
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

<!--
If this pull request resolves any GitHub issues, link them.
For information about linking a pull request to an issue, see:
https://docs.github.com/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue
-->
Resolves #NNNNN, fix apple/llvm-project#MMMMM.

<!--
Before merging this pull request, you must run the Swift continuous integration tests.
For information about triggering CI builds via @swift-ci, see:
https://github.com/apple/swift/blob/main/docs/ContinuousIntegration.md#swift-ci

Thank you for your contribution to Swift!
-->

palimondo commented:
What happened here?! This doesn't look like a proper PR to be merge into master at all!
cc @jrose-apple

zoecarver committed:
[cxx-interop] Add fix-its to apply 'safe_to_import' and 'self_contained'.

zoecarver committed:
[nfc] Add a test for the attribute fix-its.

xymus committed:
[Sema] Intro set of import filters for general use

Calling getImportedModules requires to list the desired kind of imports.
With the new kind of imports this has become cumbersome. Let's simplify
it by offering common sets of imports. Advanced call sites can still
list the desired imports explicitly.

eeckstein committed:
Swift Bridging: add the `SWIFT_IMPORT_UNSAFE` macro

It's needed for C++ bridging functions  which return types with pointers.

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedOperand` and `BridgedValue`

ahoppen committed:
[CS] Fix locator simplification with 'Member' path element

Previously, a `Member` path element in a `ConstraintLocator` was simplified to the base on which the member was accessed. This is incorrect.

xymus committed:
[Serialization] Differentiate module loading behavior for non-public imports

Differentiate `internal` and `fileprivate` imports from
implementation-only imports at the module-wide level to offer a
different module loading strategy. The main difference is for non-public
imports from a module with testing enabled to be loaded by transitive
clients.

Ideally, we would only load transitive non-public dependencies on
testable imports of the middle module. The current module loading logic
doesn't allow for this behavior easily as a module may be first loaded
for a normal import and extra dependencies would have to be loaded on
later imports. We may want to refactor the module loading logic to allow
this if needed.

rdar://106514965

xymus committed:
[Serialization] Refactor logic deciding transitive module loading logic

Refactor and centralize the logic about how implementation-only and
package-only dependencies should be loaded.

xymus committed:
[Serialization] Fix missing lookup of package-only dependencies in swiftmodule

I don't think this logic is used in practice without merge-modules.
Let's still implement it for the configurations still using
merge-modules and for general consistency.

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedSuccessor`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedVTable` and `BridgedVTableEntry`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for the bridged witness table classes

zoecarver committed:
[cxx-interop] Add fix-it for foreign reference types.

zoecarver committed:
[cxx-interop] Update fix-its to use new macro naming.

zoecarver committed:
[cxx-interop] Fix end -> nil fixit.

eeckstein committed:
Swift Bridging: remove `BridgedType` and directly use the C++ `SILType` instead

zoecarver committed:
[cxx-interop] Fixit: std.x -> CxxStdlib.x.

zoecarver committed:
[cxx-interop] Attempt to fix at -> subscript fix-it.

zoecarver committed:
[cxx-interop] Suggest 'self_contained' fixit on return type, not parent.

sophiapoirier committed:
[Variadic Generics] distinctly diagnose more types of pack expansion errors + add fixits

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedFunction`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedGlobalVar`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedBasicBlock`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedInstruction`

eeckstein committed:
SIL: move the `SILInstruction::MemoryBehavior` enum out of SILInstruction into the `swift` namespace

eeckstein committed:
Swift Bridging: remove `BridgedMemoryBehavior` and use `swift.MemoryBehavior` instead

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedArgument`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedMultiValueResult`

eeckstein committed:
Swift Bridging: remove C bridging functions for `SILDebugLocation`

eeckstein committed:
- argument

eeckstein committed:
Swift Bridging: remove `BridgedNode`

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for `BridgedBuilder`

eeckstein committed:
remove SILBridgingUtils.h

nate-chandler committed:
[NFC] Readability for isRedundantMoveValue.

Improved comments and flattened conditions.

xymus committed:
[Serialization] Do not accept packages without a name

Realign the module loading behavior with the one of the package
access-level. If the package name is an empty string, don't accept other
modules with an empty package name as being part of the same module and
don't load package dependencies in such a case.

xymus committed:
[Serialization] Try to load transitive package dependencies in the debugger

xymus committed:
[Serialization] Use the new logic for transitive dependencies in the scanner

xedin committed:
[CSFix] Diagnose an invalid member reference in ambiguous contexts

For example @objc lookup could find multiple instance members
on `AnyObject`. It should be allowed to diagnose issues with
that as non-ambiguous if all fixes have the same kind/base type.

Resolves: rdar://102412006

slavapestov committed:
IRGen: Handle PackArchetypeType in getExemplarArchetype()

bnbarham committed:
[ClangImporter] Probe for a PCH when the header is missing

The default for `ProbePrecompiled` was changed in rebranch. Set it back
to `true` to avoid breaking existing clients.

Resolves rdar://106966024.

rjmccall committed:
[NFC] Introduce a little generators library

rjmccall committed:
[NFC] Use the generators library in SILGenPoly

rjmccall committed:
[NFC] Use a generator in prolog emission to generate parameters.

This adds an assertion that we're using all of the parameters, so
pass prolog emission the number of lowered parameters to ignore.
That's easy for the callers to provide, since they do actually
still need to add function arguments for those parameters.

hyp committed:
[interop] ensure IRGen reaches code referenced through destructor invoked using the 'delete' statement in C++

zoecarver committed:
[cxx-interop][nfc] Fix test fallout.

adrian-prantl committed:
Add a verify-di-hole assertion to SILBuilder.

The verify-di-hole SILVerifier pass is very useful in catching incorrectly set
SILDebugScopes, but it can be very tedious to track down the root cause of a
verification failure. This patch replicates much of its functionality inside an
assertion in SILBuilder, which will result in a backtrace pointing directly to
where the offending scope is being set.

nate-chandler committed:
[CopyPropagation] Canonicalize moved-from values.

If a move_value is determined to be redundant and removed, take the
opportunity its removal makes available to canonicalize the moved-from
value without the obstruction of the move_value.

hyp committed:
[interop] do not print inline C++ namespaces when printing module interface

hyp committed:
[interop] NFC, bump up the experimental version in the JSON file

slavapestov committed:
IRGen: Bitcast the storage address when emitting a tuple_pack_element_addr

hborla committed:
[ConstraintSystem] Enforce TVO_CanBindToPack, and diagnose pack references outside
of pack expansion expressions.

eeckstein committed:
Swift Bridging: use C++ instead of C bridging for the optimizer

hyp committed:
Merge pull request #64493 from hyp/eng/destructor-delete-irgen

[interop] ensure IRGen reaches code referenced through destructor inv…

eeckstein committed:
SIL bridging: work around an unresolved symbol linker error

Works around  problem https://github.com/apple/swift/issues/64502

zoecarver committed:
Merge pull request #64438 from zoecarver/fixits-for-clang-types-swiftpm-2

ahoppen committed:
Merge pull request #64405 from ahoppen/ahoppen/fix-locator-simplification

[CS] Fix locator simplification with 'Member' path element

hyp committed:
Merge pull request #64497 from hyp/eng/bumpupv16

[interop] NFC, bump up the experimental version in the JSON file

ahoppen committed:
[CodeCompletion] Fix assertion failure if completing in a macro call argument

rdar://106966610

bnbarham committed:
Merge pull request #64491 from bnbarham/probe-precompiled

[ClangImporter] Probe for a PCH when the header is missing

xymus committed:
Merge pull request #64486 from xymus/import-filter-refactor

[Sema] Intro common sets of import filters to simplify calls to `getImportedModules`

hyp committed:
Merge pull request #64496 from hyp/eng/skip-inline-ns

[interop] do not print inline C++ namespaces when printing module int…

rjmccall committed:
[NFC] Extract a function to canonicalize an individual FunctionType::Param.

rjmccall committed:
[NFC] Add a function to induce a pack type from a slice of CanParams.

xymus committed:
Merge pull request #64487 from xymus/transitive-loading-refactor

[Serialization] Refactor how we decide to load transitive dependencies

eeckstein committed:
Merge pull request #64380 from eeckstein/sil-bridging

use C++ interop for SIL bridging

slavapestov committed:
IRGen: Use ConditionalDominanceScope in IRGen-generated control flow for pack iteration

nate-chandler committed:
Merge pull request #64495 from nate-chandler/copy-propagation/canonicalize-move-source

[CopyPropagation] Canonicalize moved-from values.

sophiapoirier committed:
Merge pull request #64447 from sophiapoirier/pack_expansion_error_clarity

[Variadic Generics] distinctly diagnose more types of pack expansion errors

tshortli committed:
Sema: Don't diagnose declarations more available than unavailable container.

It turns out that we must allow declarations with `introduced:` availability
nested inside of other declarations that are `unavailable` in order to
influence weak linking. Stop diagnosing declarations as being more available
than their unavailable containers and revert some changes to availability
inference that were designed to avoid creating these nestings but caused
regressions for declarations marked `@_spi_available.`

Reverts parts of https://github.com/apple/swift/pull/64310,
https://github.com/apple/swift/pull/64015, and
https://github.com/apple/swift/pull/62900.

slavapestov committed:
AST: Rename ProtocolConformance::getSubstitutions() to getSubstitutionMap() and remove ModuleDecl parameter

xymus committed:
[Serialization] Move some package-only import logic from writer to reader

Let's centralize the logic deciding if we load a transitive dependency
on the client side and have the producer write the truth in the
swiftmodule.

nate-chandler committed:
Merge pull request #64483 from nate-chandler/nfc/20230320/1/redundant-move

[NFC] Readability for isRedundantMoveValue.

hborla committed:
Merge pull request #64498 from hborla/diagnose-pack-outside-expansion-expr

[ConstraintSystem] Enforce `TVO_CanBindToPack`, and diagnose pack references outside of pack expansion expressions.

atrick committed:
PrunedLiveness: add a SILFunction argument

So that liveness can migrate to using a SILBitfield.

atrick committed:
Rename PrunedLiveness.clear() to invalidate()

Because SILBitfield cannot be cleared.

rjmccall committed:
fixup: improve generators header comment

rjmccall committed:
Allow a SILFunctionArgument to be a parameter pack.

We might want to record the *sequence* of parameter decls associated
with a parameter pack, but currently we're not doing that.

rjmccall committed:
Implement arity reabstraction for closures

artemcm committed:
[Dependency Scanning] Do not add extra search paths to the Clang's explicit PCM prebuild commands

Clang dependency scanner's output will be complete here w.r.t. the required search paths.

Resolves rdar://107018240

xedin committed:
Merge pull request #64490 from xedin/rdar-102412006

[CSFix] Diagnose an invalid member reference in ambiguous contexts

ktoso committed:
[Concurrency] Distributed actor's unownedExecutor should be optional (#64499)

tshortli committed:
Merge pull request #64515 from tshortli/revert-more-available-than-unavailable-fixes

Sema: Don't diagnose declarations more available than unavailable container

DougGregor committed:
Expand conformance macros within the conformance lookup table

The conformance lookup table is the central point of truth to
establish which protocols a nominal type conforms to. Ensure that we
expand conformance macros into that table.

Fixes rdar://106886651.

slavapestov committed:
Merge pull request #64489 from slavapestov/fix-pack-archetype-folding-irgen

IRGen: Various minor fixes

hborla committed:
[ConstraintSystem] Diagnose pack expansion expressions in non-variadic contexts.

drexin committed:
Add _pointerBitWidth platform condition (#41534)

* Add _pointerBitWidth platform condition

* Fixes after rebasing

---------

Co-authored-by: Xiaodi Wu <xiaodi.wu@gmail.com>

DougGregor committed:
[Driver] Implement inference of `-plugin-path` arguments.

This was previously implemented in the new driver; backport it to the
existing driver, which is still used by SourceKit.

Fixes rdar://106790436.

DougGregor committed:
Merge pull request #64524 from DougGregor/macro-expand-conformances-multi

Expand conformance macros within the conformance lookup table

atrick committed:
Add a BitfieldRef mini-utility.

Allows migration of existing data-structures to bitfields by
separating the state that contains the bitfield into a separate
stack-allocated container.

atrick committed:
Cleanup PrunedLiveBlocks

Use BasicBlockBitfield to record per-block liveness state. This has
been the intention since BasicBlockBitfield was first introduced.

Remove the per-field bitfield from PrunedLiveBlocks. This
(re)specializes the data structure for scalar liveness and drastically
simplifies the implementation.

This utility is fundamental to all ownership utilities. It will be on
the critical path in many areas of the compiler, including at
-Onone. It needs to be minimal and as easy as possible for compiler
engineers to understand, investigate, and debug.

This is in preparation for fixing bugs related to multi-def liveness
as used by the move checker.

DougGregor committed:
Cope with Windows paths

ktoso committed:
[Concurrency] guard diagnosis from case where Job decl might be null

meg-gupta committed:
Fix SILCombine of pointer_to_address instruction

This change ensures the insertion point of the unchecked_addr_cast is after
any copies/borrows that OSSA rauw may create.

Before this fix we could end up creating invalid SIL where operands do not
dominate its instruction.

swift-ci committed:
Merge pull request #64532 from ktoso/wip-guard-from-null-job

[Concurrency] harden diagnosis func for case where Job decl might be null

ktoso committed:
disable actor_availability on ios

DougGregor committed:
Try harder to deal with / vs. \, the third hardest problem in computer science

jckarter committed:
Merge pull request #64537 from ktoso/wip-disable-availability-test

[CI] disable actor_availability test on ios

QuietMisdreavus committed:
don't assert when a module re-exports a module and one of its symbols (#64479)

rdar://106807038

hborla committed:
Merge pull request #64526 from hborla/invalid-repeat

[ConstraintSystem] Diagnose pack expansion expressions in non-variadic contexts.

ahoppen committed:
[IDE] Remove "Begin completions" and "End completions" from test cases

These test lines weren't actually providing any value and were annoying to write. Let's jut remove them.

DougGregor committed:
Update tests yet again

nate-chandler committed:
[Test] Fully disabled test.

It's failing on more platforms than originally thought.

rdar://107052715

DougGregor committed:
Merge pull request #64541 from nate-chandler/test/20230322/1

[Test] Fully disabled test.

rjmccall committed:
Merge pull request #64517 from rjmccall/arity-reabstraction-closures

Implement arity reabstraction for closures

rintaro committed:
[Macros] Add missing scope keyword for target_link_libraries

'target_link_libraries(_swiftCSwiftPluginServer ...)' didn't have it.

swift-ci committed:
Merge pull request #64518 from artemcm/NoExtraClangPCMSearchPaths

[Dependency Scanning] Do not add extra search paths to the Clang's explicit PCM prebuild commands

slavapestov committed:
Runtime: Fix MetadataCacheKey::installInto() for variadic generics

adrian-prantl committed:
Merge pull request #64494 from adrian-prantl/dihole-assertion

Add a verify-di-hole assertion to SILBuilder.

xymus committed:
Merge pull request #64522 from xymus/access-level-import-testable

[Serialization] Load non-public dependencies of modules when testing is enabled

rintaro committed:
[CMake] Add correct sanitizer options to pure Swift libs/tools

rdar://107056349

DougGregor committed:
Merge pull request #64528 from DougGregor/old-driver-plugin-path

[Driver] Implement inference of `-plugin-path` arguments.

slavapestov committed:
Runtime: MetadataCacheKey is bit too large to pass by value

ahoppen committed:
Merge pull request #64503 from ahoppen/ahoppen/macro-call-arg-completion

[CodeCompletion] Fix assertion failure if completing in a macro call argument

swift-ci committed:
Merge pull request #64533 from meg-gupta/fixsilcombineptai

Fix SILCombine of pointer_to_address instruction

ktoso committed:
[Workaround][Concurrency] Mark Job as NOT moveonly until we fix moveonly types (#64536)

rintaro committed:
Merge pull request #64545 from rintaro/cmake-pureswift-sanitizer-rdar107056349

[CMake] Add correct sanitizer options to pure Swift libs/tools

atrick committed:
Merge pull request #64534 from atrick/liveblocks-bitfield

Cleanup PrunedLiveBlocks

rintaro committed:
Merge pull request #64543 from rintaro/macros-pluginserver-cmakefix1

[Macros] Add missing scope keyword for target_link_libraries

ahoppen committed:
Merge pull request #64506 from ahoppen/ahoppen/no-begin-end-completions

[IDE] Remove "Begin completions" and "End completions" from test cases

slavapestov committed:
Merge pull request #64547 from slavapestov/metadata-cache-key-oops

Runtime: Fix MetadataCacheKey::installInto() for variadic generics

slavapestov committed:
Merge pull request #64527 from slavapestov/protocol-conformance-cleanup

AST: Rename ProtocolConformance::getSubstitutions() to getSubstitutionMap() and remove ModuleDecl parameter

ktoso committed:
disable a few tests on iOS while we investigate

lorentey committed:
Merge pull request #64558 from ktoso/wip-disable-ios-executor

disable a few tests on iOS while we investigate
